pkg/manager/member/tidb_member_manager.go

```diff
@@ -297,6 +301,110 @@ func (tmm *tidbMemberManager) syncTiDBClientCerts(tc *v1alpha1.TidbCluster) erro
 	return tmm.certControl.Create(controller.GetOwnerRef(tc), certOpts)
 }
 
+func (tmm *tidbMemberManager) syncTiDBService(tc *v1alpha1.TidbCluster) error {
+
+	newSvc := getNewTiDBServiceOrNil(tc)
+	// TODO: delete tidb service if user remove the service spec deliberately
+	if newSvc == nil {
+		return nil
+	}
+
+	ns := newSvc.Namespace
+
+	oldSvcTmp, err := tmm.svcLister.Services(ns).Get(newSvc.Name)
+	if errors.IsNotFound(err) {
+		err = SetServiceLastAppliedConfigAnnotation(newSvc)
+		if err != nil {
+			return err
+		}
+		return tmm.svcControl.CreateService(tc, newSvc)
+	}
+	if err != nil {
+		return err
+	}
+	oldSvc := oldSvcTmp.DeepCopy()
+
+	// Adopt orphaned service by simply overriding
+	if metav1.GetControllerOf(oldSvc) == nil {
+		err = SetServiceLastAppliedConfigAnnotation(newSvc)
+		if err != nil {
+			return err
+		}
+		_, err = tmm.svcControl.UpdateService(tc, newSvc)
+		return err
+	}
+
+	equal, err := serviceEqual(newSvc, oldSvc)
+	if err != nil {
+		return err
+	}
+	annoEqual := isSubMapOf(newSvc.Annotations, oldSvc.Annotations)
+	if !equal || !annoEqual {
+		svc := *oldSvc
+		svc.Spec = newSvc.Spec
```

Comment:
The `svc.Spec.ClusterIP` should be assigned to the old one, or it will be lost.

```suggestion
		svc.Spec = newSvc.Spec
		svc.Spec.ClusterIP = oldSvc.Spec.ClusterIP
```

Reply:
Good catch, I will add some test cases for it